### DISCUSSION TITLE
Welcome to AnimatedGIF Discussions!

### DISCUSSION THREAD

**Discussion by bitbank2** [bitbank2/AnimatedGIF#81] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bitbank2** [comment]
I created this library to allow microcontrollers with limited memory to be able to decode GIF images and animations. The 'standard' libgif assumes that you have a full framebuffer to received the images while my code tries to feed it line by line to reduce the resources needed. This also has the benefit of increasing the rendering speed to SPI LCDs if you can use DMA to transmit the pixels (decoding can occur while the previous line is sent to the display). A while ago I decided to start adding 'big PC' features to my library because there's no reason why the same code can't serve for small and big systems. Many years ago I experimented with an even faster way to decode LZW streams by using the output image as the dictionary. It requires more memory, but can be tremendously faster. This weekend I decided to add this code to the library. Feel free to comment on these ideas or anything else that would benefit the users of AnimatedGIF.

**Reply by bitbank2** [reply]
I spent a good amount of time optimizing the code with the help of the Xcode profiler. Here's the latest profile running an 8-frame 160x120 animation 10,000 times on my MacBook Air M1.
![Screenshot 2024-02-19 at 1 02 25 PM](https://github.com/bitbank2/AnimatedGIF/assets/9803883/9dc70f05-ba70-4807-b396-3ac8ecf16ad4)
Trying to reduce the memory footprint slows things down, so it's only for systems with enough RAM. An example scenario is decoding a 320x240 image on an ESP32 to be written to a SPI LCD. In this case, the code needs about 54K + framesize = ~130K of RAM.

**Comment by edx-twin3** [comment]
Hi there, 
I'm afraid i haven't been able to make use of the "turbo mode", 
I'm trying to decode a 320x240 gif image, i have plenty of RAM 150k for the pTurboBuffer, the lvgl version is 9.4.0.
<img width="842" height="93" alt="image" src="https://github.com/user-attachments/assets/4a37fa4f-d2b2-472e-b3fc-d14fcd3dface" />
The buffer (px_buff) is full of 0x00
<img width="896" height="158" alt="image" src="https://github.com/user-attachments/assets/3b37a016-4045-4ef4-844b-e120d4e7d5b8" />
So i don't know what could be wrong, i tried changing the default options for these definitions with the same output (the buffer full of 0x00)
`/*GIF Defines and variables */
#define MAX_CHUNK_SIZE 512
//
// These 2 macros can be changed to limit the amount of RAM
// required by the decoder. For example, decoding 1-bit images to
// a 128x32 display will not need a max code size of 12 nor a palette
// with 256 entries
//
#define TURBO_BUFFER_SIZE 150*(1024U)`

So any help would be appreciated, by the way the normal DecodeLZW works fine.
Cheers! :1
